src/core/pointcloud/qgspointcloudlayer.cpp

```diff
@@ -474,6 +476,21 @@ void QgsPointCloudLayer::onPointCloudIndexGenerationStateChanged( QgsPointCloudD
   }
 }
 
+void QgsPointCloudLayer::onPointCloudStatisticsGenerationStateChanged( QgsPointCloudDataProvider::PointCloudStatisticsGenerationState state )
```

Comment:
Rather than having data providers trigger stats calculation, it will be better to have all that contained in QgsPointCloudLayer... so that once we have the layer indexed, we check whether the stats are available, and if not, we trigger the generation in QgsPointCloudLayer as a next step.

Reply:
This is inconvenient since not point cloud layer types will do the indexing, and for that no signals will be fired at all, I am trying to avoid having to add specific code for each type of layers: EPT, PDAL indexed layers, COPC, EPTs with no statistics.
I chose to trigger the statistics calculation inside of the data provider because it is easier to figure out when to trigger the statistics calculation rather that trying to figure that out in the point cloud layer.